odoo_module_migrate/migration_scripts/migrate_130_140.py

```diff
@@ -1,12 +1,85 @@
 # Copyright (C) 2020 - Today: Iván Todorovich
+# Copyright (C) 2020 - Today: Simone Rubino
 # @author: Iván Todorovich (https://twitter.com/ivantodorovich)
+# @author: Simone Rubino (https://github.com/Daemo00)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from pathlib import Path
+import lxml.etree as et
 
-_TEXT_ERRORS = {
-    ".xml": {
-        "<act_window(\n|.|\t)*":
-        "act_window tag has been deprecated. Use <record> instead",
-        "<report(\n|.|\t)*":
-        "report tag has been deprecated. Use <record> instead",
-    }
-}
+
+def _reformat_file(file_path: Path):
+    """Reformat `file_path`.
+
+    Substitute `act_window` and `report` tag with `record` tag.
+    Note that:
+    - `id` attribute is kept in the `record` tag;
+    - in `report` tag:
+      - `string` has been renamed to `name`;
+      - `name` has been renamed to `report_name`;
+    - other attributes are assigned to respective fields
+    """
+    file_path = str(file_path.resolve())
+    parser = et.XMLParser(remove_blank_text=True)
```

Comment:
I think the `remove_blank_text` option is causing a too big diff when migrating. I tried removing it and the diff is much clearer and makes more sense. I don't know if there are other situations where this makes sense.

Also, this parser always removes the `<?xml version="1.0" encoding="utf-8" ?>` line at the top. Is that expected/normal?

Reply:
Hi, thanks for having a look!
I added the creation of the encoding line, I actually never noticed it was missing.

About the `remove_blank_text` attribute, I tried to remove it and the migrated XML is:
```
<?xml version='1.0' encoding='UTF-8'?>
<odoo>
    <record id="act_window_example" model="ir.actions.act_window"><field name="name">Example</field><field name="res_model">account.move.line</field><field name="binding_model">account.move</field><field name="view_mode">tree</field></record>
    <record id="report_example" model="ir.actions.report"><field name="report_name">Name</field><field name="name">String</field></record>
</odoo>
```
so I'd rather keep it.
Maybe the XMLs that are producing a big diff really have to be reformatted?

Reply:
Ok.
Although, in this case, as there is pre-commit in v14 OCA repos, it would end up bringing it back to a prettier format.
The problem with the current `remove_blank_text` is that it ends up reformatting entire files, instead of just the part that needs to be changed, which causes bigger diffs in migrations...
Not a huge problem, though, good job!